app/src/main/java/com/example/bankdetails/adapter/BanksAdapter.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.bankdetails.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.navigation.findNavController
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.bankdetails.model.BankInfo
+import com.example.bankdetails.databinding.ItemBanksBinding
+import com.example.bankdetails.fragment.BanksFragmentDirections
+
+class BanksAdapter :
+    ListAdapter<BankInfo, BanksAdapter.BanksListViewHolder>(diffUtils) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BanksListViewHolder {
+        val binding =
+            ItemBanksBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return BanksListViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: BanksListViewHolder, position: Int) {
+        holder.bind(getItem(position))
+    }
+
+    inner class BanksListViewHolder(private val binding: ItemBanksBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(item: BankInfo) {
+            binding.bankData = item
+            itemView.setOnClickListener {
+                binding.root.findNavController().navigate( BanksFragmentDirections.navigateToBankDetailsFragment(
```

Comment:
The adapter should only take data and bind the data to the view. If you're taking action on click it would suggest that this adapter can only be used in one place. The on click should return a callback and allow the activity/fragment to decide what to do.

Reply:
Done